datastore/src/main/java/org/jboss/pnc/datastore/repositories/BuildRecordRepositoryImpl.java

```diff
@@ -310,6 +317,57 @@ public int countAllBuildRecordInsightsNewerThanTimestamp(Date lastupdatetime) {
         return count.intValue();
     }
 
+    @Override
+    public List<Tuple> getBuildsProducingArtifactDependencies(Base32LongID buildId) {
```

Comment:
This query can be much simpler and faster.

```java
        CriteriaBuilder cb = entityManager.getCriteriaBuilder();
        CriteriaQuery<Tuple> query = cb.createQuery(Tuple.class);

        Root<BuildRecord> builds = query.from(BuildRecord.class);
        SetJoin<BuildRecord, Artifact> dependencyArtifacts = builds.join(BuildRecord_.dependencies);

        var dependantBuildRecordIds = dependencyArtifacts.get(Artifact_.buildRecord).get(BuildRecord_.id);
        query.multiselect(dependantBuildRecordIds, cb.count(dependencyArtifacts.get(Artifact_.id)).as(Integer.class));
        query.where(cb.equal(builds.get(BuildRecord_.id), buildId));

        query.groupBy(dependantBuildRecordIds);

        return entityManager.createQuery(query).getResultList();
```

Reply:
done